snuba/web/rpc/__init__.py

```diff
@@ -355,13 +384,13 @@ def __after_execute(self, in_msg: Tin, out_msg: Tout, error: Exception | None) -
                 tags=self._timer.tags,
             )
 
-        if output_metrics_error is not None:
-            self.metrics.increment("metrics_failure")
+        if routing_strategy_after_execute_error is not None:
+            self.metrics.increment("after_execute_failure")
             sentry_sdk.capture_message(
-                f"Error in routing strategy output metrics: {output_metrics_error}"
+                f"Error in routing strategy after execute: {routing_strategy_after_execute_error}"
             )
             if settings.RAISE_ON_ROUTING_STRATEGY_FAILURES:
-                raise output_metrics_error
+                raise routing_strategy_after_execute_error
```

Comment:
this exception handling should probably be taken care of in the routing strategies